### PR TITLE
systemd: generate stable machine IDs for VMWare guests

### LINKED ISSUE #1536: Journal is recreated with a different machine ID after reboot on VMWare
**Image I'm using:**
Bottlerocket vmware-dev

**What I expected to happen:**
Systemd shouldn't use the random machine id generator for the journal file

**What actually happened:**
Systemd is using the random machine id generator for the journal file

**How to reproduce the problem:**
- Launch a vmware-dev host
- Reboot
- Run `journalctl --list-boots` from `sheltie`, and only the last boot's logs will appear

### PR DESCRIPTION
**Pull request by arnaldo2792**
**Issue number:**
#1536

**Description of changes:**

```
5c751ff2 systemd: generate stable machine IDs for VMWare guests
```

Systemd already has functionality to generate stable machine ids based on the virtualization type. This patch extends the functionality to check for the VMWare value as well.

**Testing done:**

In AWS dev, ecs, vmware-dev:

- `systemctl status` didn't show any failures
- `journalctl -p3` didn't show any unexpected errors
- Verified that XEN, KVM (x86_64 and aarch64) and VMWare hosts have a stable machine id after reboots and shutdowns:

**AWS ECS c4.large (XEN)**

```shell
bash-5.0# journalctl --list-boots
-2 2ed3bba8aac349919ac3985e72905281 Wed 2021-04-28 22:58:10 UTC—Wed 2021-04-28 23:02:03 UTC
-1 5bd4344b80e04ac5b0693b46e9b10047 Wed 2021-04-28 23:02:23 UTC—Wed 2021-04-28 23:05:42 UTC
 0 8bc4237e9e1e4592a7c8a20655df70ff Wed 2021-04-28 23:53:36 UTC—Wed 2021-04-28 23:54:36 UTC
```

**AWS ECS m5.xlarge (KVM)**

```shell
bash-5.0# journalctl --list-boots
-2 98f91b7d5bdd46d78e218969bf5a63c2 Wed 2021-04-28 22:52:33 UTC—Wed 2021-04-28 22:55:50 UTC
-1 7c4015c7155d4349ad894d9fc72885ec Wed 2021-04-28 22:55:56 UTC—Wed 2021-04-28 22:57:21 UTC
 0 0eaf0223bc4d43bf881d828f2b1f7eec Wed 2021-04-28 23:01:34 UTC—Wed 2021-04-28 23:02:45 UTC
```

**AWS ECS t4g.medium (KVM aarch64)**

```
bash-5.0# journalctl --list-boots
-2 64df9034e0184d74944f1a48e33ac31a Wed 2021-04-28 22:53:08 UTC—Wed 2021-04-28 22:56:09 UTC
-1 350ce32cf0f04fecac8d810c62842cdd Wed 2021-04-28 22:56:13 UTC—Wed 2021-04-28 22:57:21 UTC
 0 6ffa095e34a547a183e5458e9072f042 Wed 2021-04-28 23:01:33 UTC—Wed 2021-04-28 23:04:04 UTC
```

**VMWare**
Screenshot from 2021-04-28 17-28-13

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: packages/systemd/9009-machine-id-setup-generate-stable-ID-under-VMWare.patch [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
++                        /* If we're running under Xen, check for a UUID from
++                         * the hypervisor. */
+ 
+                         if (id128_read("/sys/hypervisor/uuid", ID128_UUID, ret) >= 0) {
+                                 log_info("Initializing machine ID from Xen UUID.");
+                                 return 0;
+                         }
++                } else if (virtualization_type == VIRTUALIZATION_VMWARE) {
++                        /* If we're running under VMWare, check for a machine ID
++                         * passed in via --uuid */
```
Comment (bcressey) [inline packages/systemd/9009-machine-id-setup-generate-stable-ID-under-VMWare.patch]: I don't think this comment is accurate for VMWare.

File: packages/systemd/9009-machine-id-setup-generate-stable-ID-under-VMWare.patch [hunk 2]
```diff
@@ -0,0 +1,66 @@
+From f18ea74a0a62436849cf26ba9b8d5ab78d4f61a5 Mon Sep 17 00:00:00 2001
+From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
+Date: Wed, 28 Apr 2021 22:23:23 +0000
+Subject: [PATCH 9009/9009] machine-id-setup: generate stable ID under VMWare
```
Comment (bcressey) [inline packages/systemd/9009-machine-id-setup-generate-stable-ID-under-VMWare.patch]: This should get merged into a single patch with 9004, so we don't have two patches touching the same part of upstream code. Don't worry about preserving the attribution for me.

File: packages/systemd/9009-machine-id-setup-generate-stable-ID-under-VMWare.patch (hunk 1 above)
Comment (zmrow) [reply packages/systemd/9009-machine-id-setup-generate-stable-ID-under-VMWare.patch]: Also - the company styles it as `VMware`, let's follow suit. :)

**Comment by arnaldo2792** [comment]
- Use one single patch to generate stable machine IDs for XEN and VMware
- Fix comment regarding how the uuid is selected

**Comment by arnaldo2792** [comment]
- Fixed comment and commit message to use VMware instead of VMWare

**Review by zmrow — APPROVED** [review]
:yin_yang:

**Review by tjkirch — APPROVED** [review]

**Review by bcressey — APPROVED** [review]

**Pull request merged**